lib/scholar/naive_bayes/gaussian_naive_bayes.ex

```diff
@@ -0,0 +1,442 @@
+
+        {new_mu, new_var, new_n} =
+          update_mean_variance_init(
+            theta[[i, 0..-1//1]],
+            var[[i, 0..-1//1]],
+            x,
+            sample_weights,
+            mask,
+            sample_weights_flag
+          )
+
+        {new_mu, new_var} =
+          if class_count[[i]] != 0,
+            do:
+              update_mean_variance(
+                class_count[[i]],
+                new_n,
+                new_mu,
+                new_var,
+                theta[[i, 0..-1//1]],
+                var[[i, 0..-1//1]]
+              ),
+            else: {new_mu, new_var}
+
+        new_theta = Nx.put_slice(theta, [i, 0], Nx.broadcast(new_mu, {1, num_features}))
+        new_var = Nx.put_slice(var, [i, 0], Nx.broadcast(new_var, {1, num_features}))
+        class_count = Nx.indexed_add(class_count, Nx.reshape(i, {1, 1}), Nx.reshape(n_i, {1}))
+
+        {i + 1, x, y, sample_weights, classes, sample_weights_flag, new_theta, new_var,
+         class_count}
+      end
+
+    final_var = final_var + eps
+
+    class_priors =
+      if priors_flag, do: class_priors, else: final_class_count / Nx.sum(final_class_count)
+
+    %__MODULE__{
+      theta: final_theta,
+      var: final_var,
+      class_count: final_class_count,
+      class_priors: class_priors,
+      classes: classes,
+      epsilon: eps
+    }
+  end
+
+  defnp update_mean_variance_init(mu, var, x, sample_weights, mask, sample_weights_flag) do
+    {num_samples, _num_features} = Nx.shape(x)
+
+    cond do
+      num_samples == 0 ->
+        {mu, var, 0}
+
+      sample_weights_flag == false ->
+        new_n = num_samples
+        new_mu = mean_masked(x, mask)
+        new_var = mean_masked((x - new_mu) ** 2, mask)
+        {new_mu, new_var, new_n}
+
+      true ->
+        new_n = Nx.sum(sample_weights * mask)
+        new_mu = mean_weighted_masked(x, mask, sample_weights)
+        new_var = mean_weighted_masked((x - new_mu) ** 2, mask, sample_weights)
+        {new_mu, new_var, new_n}
+    end
+  end
+
+  defnp update_mean_variance(past_n, new_n, new_mu, new_var, mu, var) do
+    total_n = past_n + new_n
+    total_mu = (new_n * new_mu + past_n * mu) / total_n
+
+    past_ssd = past_n * var
+    new_ssd = new_n * new_var
+    total_ssd = past_ssd + new_ssd + new_n * past_n / total_n * (mu - new_mu) ** 2
+    total_var = total_ssd / total_n
+
+    {total_mu, total_var}
+  end
+
+  defnp joint_log_likelihood(
+          %__MODULE__{class_priors: class_priors, var: var, theta: theta},
+          x
+        ) do
+    pi = 3.14159265359
+    joint = Nx.log(class_priors)
+    {num_classes, num_features} = Nx.shape(theta)
+    {samples_x, _features_x} = Nx.shape(x)
+
+    n1 =
+      (-0.5 * Nx.sum(Nx.log(2.0 * pi * var), axes: [1]))
+      |> Nx.reshape({:auto, 1})
+      |> Nx.broadcast({num_classes, samples_x})
+
+    x_broadcast =
+      Nx.reshape(x, {1, samples_x, num_features})
```

Comment:
`num_features` and `_features_x` are the same, right?
```suggestion
      Nx.new_axis(x, 0)
```